### DISCUSSION TITLE
How to develop an interactive changing pages in Avalonia?

### DISCUSSION THREAD

**Discussion by aleh4d** [AvaloniaUI/Avalonia#9333] [Q&A] [answered]
Hi Avalonia developers, thank you for the goo work,
We want to develop an app that will be an interactive book, that means changing from one page with controls to another page many times.
But I see in examples there is only one Window object, and all the content is placed inside,, and there is no way to smoothly change to another window inside application without closing the previous window?
What will be the best way to develop an interactive book with controls and with many pages in Avalonia? Can you provide some examples or code hints?

**Comment by FoggyFinder** [comment]
You don't have to change Windows. Use UserControl instead that would represent a "page".

**Comment by aleh4d** [comment]
Does it need to dynamically create in code a new user control and put it as a child control for the Window object each time a new page is needed? 
Or it would be better to put many user controls in XAML and just make those controls visible/invisible when the pange needs to be shown?
Which way would be tetter? (any other suggestions in case a mult-page app needs to be developed which needs to go from one pave to anogher? (like in web going from one page to another with links)

**Comment by FoggyFinder** [comment]
You can start here:

[Tutorial](https://avaloniaui.net/docs/tutorial/)

**Comment by Killjoyer** [comment]
> You can start here:
> 
> [Tutorial](https://avaloniaui.net/docs/tutorial/)

It's 404.  
I googled "Avalonia switching window without opening new one" and got here. And lots of tutorial and quickstart links lead to 404, even the one from [docs.avaloniaui.net](https://docs.avaloniaui.net/docs/controls/window). What's wrong with this?

**Comment by FoggyFinder** [comment, marked as answer]
> > You can start here:
> > [Tutorial](https://avaloniaui.net/docs/tutorial/)
> 
> It's 404.
> I googled "Avalonia switching window without opening new one" and got here. And lots of tutorial and quickstart links lead to 404, even the one from [docs.avaloniaui.net](https://docs.avaloniaui.net/docs/controls/window). What's wrong with this?

@Killjoyer Documentation was moved to another site. The proper reference now is 

[ToDo List App](https://docs.avaloniaui.net/tutorials/todo-list-app)

**Reply by timunie** [reply]
if someone is looking for v11 tutorial: https://docs.avaloniaui.net/docs/next/tutorials/todo-list-app/

**Reply by alexzandros** [reply]
Links don't work

**Reply by FoggyFinder** [reply]
@alexzandros https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/CompleteApps/SimpleToDoList

**Comment by timunie** [comment]
Converted this into Q&A as this is neither a bug nor a feature request. If you disagree please let us know.

**Reply by bsampica** [reply]
Link for this is dead again it seems - The correct URL is here:

https://docs.avaloniaui.net/docs/tutorials/todo-list-app/.  

working august 2023

**Reply by timunie** [reply]
the correct link for 11.0 (current version of Avalonia) is : https://docs.avaloniaui.net/docs/next/tutorials/todo-list-app/
